debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -1275,7 +1269,7 @@ protected String resolveCatalogName(String catalogName) {
      * Given a database object name, creates a pattern, escaping any special characters such that the pattern matches
      * the name itself.
      */
-    protected String createPatternFromName(String name) {
+    protected String createPatternFromName(String name, String databaseSearchEscapeCharacter) {
```

Comment:
I'd rename `databaseSearchEscapeCharacter` to `searchEscapeCharacter`:
1. It aligns with the `DatabaseMetadata` method name.
2. The actual value is defined not so by the database as by the driver (technically any SQL query can use its own escape character, and the driver uses one of them).